waspc/data/Generator/templates/react-app/src/auth/forms/Auth.jsx

```diff
@@ -0,0 +1,300 @@
+    backgroundColor: '$brandAccent',
+    borderColor: '$brandAccent',
+  },
+  transitionTimingFunction: 'cubic-bezier(0.4, 0, 0.2, 1)',
+  transitionDuration: '100ms'
+})
+
+const googleSignInUrl = `${config.apiUrl}{= googleSignInPath =}`
+const gitHubSignInUrl = `${config.apiUrl}{= gitHubSignInPath =}`
+
+const Auth = ({ isLogin, appearance, logo, socialLayout }) => {
+  const history = useHistory()
+
+  const [usernameFieldVal, setUsernameFieldVal] = useState('')
+  const [passwordFieldVal, setPasswordFieldVal] = useState('')
+
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    try {
+      if (!isLogin) {
+        await signup({ username: usernameFieldVal, password: passwordFieldVal })
+      }
+      await login (usernameFieldVal, passwordFieldVal)
+
+      setUsernameFieldVal('')
+      setPasswordFieldVal('')
+
+      // Redirect to configured page, defaults to /.
+      history.push('{= onAuthSucceededRedirectTo =}')
+    } catch (err) {
+      console.log(err)
+      window.alert(errorMessage(err))
+    }
+  }
+
+  // TODO(matija): this is called on every render, is it a problem?
+  // If we do it in useEffect(), then there is a glitch between the default color and the
+  // user provided one.
+  const customTheme = createTheme(appearance)
+
+  const cta = isLogin ? 'Log in' : 'Sign up'
+  const title = isLogin ? 'Log in to your account' : 'Create a new account'
+
+  const socialButtonsDirection = socialLayout == 'vertical' ? 'vertical' : 'horizontal'
+
+  return (
+    <Container className={customTheme}>
+      <div>
+        { logo && (
+          <img style={logoStyle} src={logo} alt='Your Company' />
+        )}
+        <HeaderText>{title}</HeaderText>
+      </div>
+
+      {=# isExternalAuthEnabled =}
+        <SocialAuth>
+          <SocialAuthLabel>{cta} with</SocialAuthLabel>
+          <SocialAuthButtons gap='large' direction={socialButtonsDirection}>
+            {=# isGoogleAuthEnabled =}
+              <SocialButton href={googleSignInUrl}><SocialIcons.Google/></SocialButton>
+            {=/ isGoogleAuthEnabled =}
+
+            {=# isGitHubAuthEnabled =}
+              <SocialButton href={gitHubSignInUrl}><SocialIcons.GitHub/></SocialButton>
+            {=/ isGitHubAuthEnabled =}
+          </SocialAuthButtons>
+        </SocialAuth>
+      {=/ isExternalAuthEnabled =}
+      
+      {=# isBothExternalAndUsernameAndPasswordAuthEnabled =}
+        <OrContinueWith>
+          <OrContinueWithLineContainer>
+            <OrContinueWithLine/>
+          </OrContinueWithLineContainer>
+          <OrContinueWithTextContainer>
+            <OrContinueWithText>Or continue with</OrContinueWithText>
+          </OrContinueWithTextContainer>
+        </OrContinueWith>
+      {=/ isBothExternalAndUsernameAndPasswordAuthEnabled =}
+      
+      {=# isUsernameAndPasswordAuthEnabled =}
+        <UserPassForm onSubmit={handleSubmit}>
+          <FormItemGroup>
+            <FormLabel>Username</FormLabel>
```

Comment:
We should consider i18n and enable users to change label text, but maybe not for v1.

Reply:
yes, for sure! We can do sth like this: https://supabase.com/docs/guides/auth/auth-helpers/auth-ui#custom-labels

I opened an issue for it: https://github.com/wasp-lang/wasp/issues/1104